staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -353,6 +386,19 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 		token: cred.Status.Token,
 	}
 	if cred.Status.ClientKeyData != "" && cred.Status.ClientCertificateData != "" {
+		// Check cert expiration time.
+		certBlock, _ := pem.Decode([]byte(cred.Status.ClientCertificateData))
+		if certBlock == nil {
+			return errors.New("can't parse client certificate returned by exec plugin as PEM block")
+		}
+		x509Cert, err := x509.ParseCertificate(certBlock.Bytes)
+		if err != nil {
+			return fmt.Errorf("failed parsing x509 client certificate returned by exec plugin: %v", err)
+		}
+		if time.Now().After(x509Cert.NotAfter) {
```

Comment:
are we already checking time client-side elsewhere? this makes us vulnerable to client clock skew

Reply:
I don't know if we're checking it elsewhere. It should be kube-apiserver's responsibility to verify cert validity on incoming requests. Could you clarify how this makes us vulnerable?